### PR TITLE
JAMES-3311 Support concurrencly in ImapChannelUpstreamHandler

### PR DESCRIPTION
**Pull request by blackheaven**
#3576 

I do not manage to reproduce concurrency

### CODE REVIEW

**Comment by chibenwa** [comment]
Should we write a simple custom client using reactor-netty to achieve this?

**Comment by chibenwa** [comment]
`[9408ff970d91405a551543fea7315d8f053092aa] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:check (default) on project james-server-testing: You have 1 Checkstyle violation. -> [Help 1]`

`[9408ff970d91405a551543fea7315d8f053092aa] [ERROR] src/main/java/org/apache/james/utils/TestIMAPClient.java:[33,8] (imports) UnusedImports: Unused import - org.apache.commons.net.imap.IMAPCommand.`

**Comment by blackheaven** [comment]
it seems to work with FETCH

File: server/container/guice/guice-common/src/test/java/org/apache/james/MailsShouldBeWellReceived.java [hunk 1]
```diff
@@ -257,6 +259,38 @@ default void oneHundredMailsShouldBeWellReceived(GuiceJamesServer server) throws
...
+    @Test
+    default void oneHundredFetchesShouldBePerformableConcurrently(GuiceJamesServer server) throws Exception {
+        server.getProbe(DataProbeImpl.class).fluent()
+                .addDomain(DOMAIN)
+                .addUser(JAMES_USER, PASSWORD);
+
+        MailboxProbeImpl mailboxProbe = server.getProbe(MailboxProbeImpl.class);
+        mailboxProbe.createMailbox("#private", JAMES_USER, DefaultMailboxes.INBOX);
+
+        int fetchAttemps = 100;
```
Comment (LanKhuat) [inline server/container/guice/guice-common/src/test/java/org/apache/james/MailsShouldBeWellReceived.java:271]: s/Attemps/Attempts

File: server/container/guice/guice-common/src/test/java/org/apache/james/MailsShouldBeWellReceived.java [hunk 2]
```diff
@@ -257,6 +259,38 @@ default void oneHundredMailsShouldBeWellReceived(GuiceJamesServer server) throws
...
+        String message = Resources.toString(Resources.getResource("eml/htmlMail.eml"), StandardCharsets.UTF_8);
+        try (TestIMAPClient imapClient = new TestIMAPClient()) {
+            imapClient.connect(JAMES_SERVER_HOST, server.getProbe(ImapGuiceProbe.class).getImapPort())
+                .login(JAMES_USER, PASSWORD)
+                .select(TestIMAPClient.INBOX);
+
+            appendUniqueMessage(imapClient, TestIMAPClient.INBOX, message);
+
+            Flux.range(0, fetchAttemps)
```
Comment (LanKhuat) [inline server/container/guice/guice-common/src/test/java/org/apache/james/MailsShouldBeWellReceived.java:280]: idem

**Comment by blackheaven** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

unrelated

**Comment by chibenwa** [comment]
Issue solved

**Pull request closed without merging**